client/client.go

```diff
@@ -1754,11 +1754,11 @@ func (c *Client) deriveToken(alloc *structs.Allocation, taskNames []string, vcli
 	var resp structs.DeriveVaultTokenResponse
 	if err := c.RPC("Node.DeriveVaultToken", &req, &resp); err != nil {
 		c.logger.Printf("[ERR] client.vault: DeriveVaultToken RPC failed: %v", err)
-		return nil, fmt.Errorf("DeriveVaultToken RPC failed: %v", err)
+		return nil, structs.NewWrappedServerError(resp.Error)
```

Comment:
This should return the err not the resp error

Reply:
Also should it be marked as a server error? It is likely there is no valid server which is where the error is coming from since all errors other than network are returned via the `resp.Error`

Reply:
You are right, this should not be marked as such, will fix